### PR TITLE
#63_using_implemented_model 01

### LINKED ISSUE #63: Using _ImplementedModel_01
## 取り組むトレーニングの概要
「ビギナー：sklearnを用いた実装済みモデルの利用」にて指定されている実装済みモデルを利用し、単回帰とPCAの処理を実行する。

## 取り組む期間
5/5 ~ 5/6
## 獲得を目指すスキル
1. irisデータセットに対してPCAの処理を適用できる。
2. Boston house-pricesに対して線形回帰の処理を適用できる。

## Action list
 - [x] sklearnの単回帰のサンプルコードを写経する。
 - [x] sklearnのPCAのサンプルコードを写経する。
 - [x] ミニレポートを提出する。

### PR DESCRIPTION
**Pull request by oirom**
## 取り組んだトレーニングの概要
「ビギナー：sklearnを用いた実装済みモデルの利用」にて指定されている実装済みモデルを利用し、単回帰とPCAの処理を実行する。

close #63 

## Action list
- sklearnの単回帰のサンプルコード(diabetes)を写経した。
- sklearnの単回帰のサンプルコードを参考に単回帰をboston datasetに対して適用した。
- sklearnのPCAのサンプルコード(iris)を写経した。

## レビューして欲しいポイント
単回帰に取り組むのは初めてだった。一度diabetes datasetについて写経したのち、復習の意味も兼ねてboston datasetに対しても単回帰を適用した。

## レビュワーへの質問
単回帰においてVariance scoreの計算がありましたが、調べたところVariance scoreが1.0に近いほど高い精度であるとの記事がありました。Varianceとは分散のことと認識しています。これは、PCAにおける主成分の分散（分散共分散行列の固有値）をPCA適用前のデータの分散の和で割った「寄与率」と同様のものでしょうか？

## 感想
単回帰についてはコードから何を行っているのかを想像しやすく、結果も予想通りのプロットが得られた。PCAについても前処理のビギナー&常人にて複数回写経済みだったのでスムーズに作業できた。

## 次に取り組むトレーニング
『モデル 常人：Kerasを用いたNNモデルの構築』に取り組みたい。ニューラルネットワークを構築してみたいです。

### CODE REVIEW

**Comment by ae14watanabe** [comment]
> Variance score〜これは、PCAにおける主成分の分散（分散共分散行列の固有値）をPCA適用前のデータの分散の和で割った「寄与率」と同様のものでしょうか？

Explained variance scoreってやつですよね。分散なのは間違い無いですけど、PCAのやつとは別物じゃないですかね。
自分もこの評価指標知らなかったので、[ここ](https://funatsu-lab.github.io/open-course-ware/basic-theory/accuracy-index/#explained-variance-score)を見ただけの判断ですが、どうやら残差（データと予測値の差）の分散を使って評価しましょうって話ですね。PCAで言ってるのはデータを次元削減した値の分散なので、「何の」分散を測っているかが違います。

**Review by ae14watanabe — APPROVED** [review]
お疲れ様です！sklearnのモデルを呼び出すのはもうお手の物ではないでしょうか？メニューで獲得したいスキルから考えると十分ですのでApproveします。これからもファイトです！

**Review by takuro-Ishida — APPROVED** [review]
お疲れ様です！
プログラムの内容も確認してIssueに上がっているスキルを獲得できていると判断したのでApproveします！
いろんなデータセットを使ったり, sklearnからモデルを呼び出せるとか,いろんなデータセットを知っているというのはとても大事だと思うのでとても良いと思います！

この調子で頑張ってください!

**Comment by oirom** [comment]
@takuro-Ishida コメントありがとうございます！各種モデルの利用やデータセットの操作を身に着けられるよう努力します！
